city_scrapers/spiders/newnj_nbe.py

```diff
@@ -0,0 +1,142 @@
+    timezone = "America/New_York"
+    start_urls = ["https://www.nps.k12.nj.us/board-of-education/meetings/"]
+
+    def parse(self, response):
+        """
+        Most of the meetings details are provided in a detail page.
+        Though the each meetings' location address varies and is taken
+        from the main page and passed as a meta value to the followed
+        callback method.
+        """
+        meetings_table = response.css(
+            ".su-table.su-table-alternate table tbody tr:not(:first-child)"
+        )
+        for item in meetings_table:
+            location_cell = item.css("td:nth-child(4)::text").getall()
+            detail_link = item.css("td a::attr(href)").extract_first()
+
+            yield response.follow(
+                detail_link,
+                callback=self._parse_detail,
+                meta={"location_cell": location_cell},
+            )
+
+    def _parse_detail(self, response):
+        location_cell = response.meta["location_cell"]
+
+        date = self._parse_date(response)
+        start_time, end_time = self._parse_start_end_time(response)
+
+        meeting = Meeting(
+            title=self._parse_title(response),
+            description="",
+            classification=BOARD,
+            start=self._gen_datetime(date, start_time),
+            end=self._gen_datetime(date, end_time),
+            all_day=False,
+            time_notes="",
+            location=self._parse_location(response, location_cell),
+            links=self._parse_links(response),
+            source=response.url,
+        )
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _parse_title(self, item):
+        title = item.css("h1.entry-title::text").extract_first()
+        return title
+
+    def _gen_datetime(self, date, time_obj):
+        if time_obj is None:
+            time_obj = time(0, 0)  # Midnight
+
+        return datetime.combine(date, time_obj)
+
+    def _parse_start_end_time(self, item):
+        time_str = item.css(".su-table table tr:nth-child(2) td:nth-child(2)::text")[
+            0
+        ].get()
+        if time_str:
+            time_str = time_str.strip()
+            start_time_str, end_time_str = time_str.split(" - ")
+
+            start_time = datetime.strptime(start_time_str, "%I:%M %p").time()
+            end_time = datetime.strptime(end_time_str, "%I:%M %p").time()
+
+            return start_time, end_time
+        return None, None
+
+    def _parse_date(self, item):
+        date_str = item.css(
+            ".su-table.su-table-alternate table tr td:nth-child(2)::text"
+        )[0].get()
+
+        date_obj = datetime.strptime(date_str, "%m/%d/%Y")
+
+        return date_obj
+
+    def _parse_location(self, item, location_cell):
+        if "Virtual" in location_cell:
+            return {
+                "address": "Virtual",
+                "name": "Virtual",
+            }
```

Comment:
**praise:** Good stuff. Thanks for adding this extra handling for virtual meetings ✨